data-prepper-plugins/kafka-plugins/src/main/proto/org/opensearch/dataprepper/plugins/kafka/KafkaBufferMessage.proto

```diff
@@ -16,4 +16,13 @@ message BufferData {
    * is unencrypted data.
    */
   bytes data = 2;
+
```

Comment:
I think this structure have version field to support changes like this for backward compatibility.

Reply:
it should be mandatory and should change whenever the definition of the protobuf message structure changes. This PR is changing the existing structure to a new structure. What happens if writer (producer) to kafka writes one version and the reader of kafka is a different dataprepper version with modified structure?

Reply:
Protobuf has built-in support for this. A Data Prepper running with this structure will be ok with the old data because it is marked as optional.

https://protobuf.dev/programming-guides/proto3/#updating

Reply:
I see. How does this code handle if the protobuf message doesn't have the new fields? I understand the protobuf libraries handle it gracefully, don't the code need to handle it as well?

Reply:
Right now this code is only adding the new messages as noted in the description. I will have a follow-on PR that will use the fields (if provided) to decrypt the data. If not provided, then we will attempt to decrypt with the current key.